### ISSUE TITLE
False positive attribute-defined-outside-init in Python 3.5.2

### ISSUE THREAD

**Issue by glennmatthews** [PyCQA/pylint#3037]
### Steps to reproduce

Use the following simple file:

```python
"""Demo."""

class Foo:
    """Show the issue."""

    def __init__(self):
        "Instantiate the class"
        self.attr = None
```

### Current behavior

`attribute-defined-outside-init` warning is reported under Python 3.5.2 but not under Python 3.6.2:

```
> python --version
Python 3.5.2
> pylint test.py
************* Module test
test.py:3:0: R0903: Too few public methods (0/2) (too-few-public-methods)
test.py:8:8: W0201: Attribute 'attr' defined outside __init__ (attribute-defined-outside-init)
```

```
> python --version
Python 3.6.2
> pylint test.py
************* Module test
test.py:3:0: R0903: Too few public methods (0/2) (too-few-public-methods)
```

### Expected behavior

No warning under either Python version

### pylint --version output

```
> pylint --version
pylint 2.3.1
astroid 2.2.5
```

**Comment by PCManticore** [comment]
Thanks for reporting an issue. I'm afraid I can't reproduce it though, tested with:
```
__main__.py 2.3.1
astroid 2.2.5
Python 3.5.2 (default, Sep  4 2018, 11:08:48)
```
There must be something going on with your `pylint` version on Python 3.5.2.
